utils/actonmon

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python3
+
+
+import argparse
+from functools import partial
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import json
+import socket
+import time
+
+class MonSock:
+    def __init__(self, address):
+        self.address = address
+        self.sock = None
+
+    def connect(self):
+        self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        try:
+            self.sock.connect(self.address)
+        except:
+            pass
+
+    def mon(self, retries=5):
+        if self.sock is None:
+            self.connect()
+
+        try:
+            self.sock.send(b"WTS")
+            res = self.sock.recv(20000)
+            print(res)
+            return json.loads(res.decode("utf-8"))
+        except Exception as exc:
+            self.connect()
+            if retries > 0:
+                self.mon(retries-1)
+            else:
+                raise ConnectionError("Unable to get data from acton rts")
+
+
+class PromHTTPRequestHandler(BaseHTTPRequestHandler):
+    def __init__(self, sock_addr, *args, **kwargs):
+        self.ms = MonSock(sock_addr)
+        # BaseHTTPRequestHandler calls do_GET **inside** __init__ !!!
+        # So we have to call super().__init__ after setting attributes.
+        super().__init__(*args, **kwargs)
+
+    def do_GET(self):
+        try:
+            data = self.ms.mon()
+        except ConnectionError:
+            self.send_response(503)
+            return
+
+        self.send_response(200)
+        self.end_headers()
+
+        conts_buckets = [k for k in data['wt']['0'] if k.startswith('conts_') and not k.endswith(("_count", "_sum"))]
+        bkeep_buckets = [k for k in data['wt']['0'] if k.startswith('bkeep_') and not k.endswith(("_count", "_sum"))]
+
+        total = {k: 0 for k in data['wt']['0']}
+
+        self.wfile.write(b'# HELP acton_rts_sleeps Number of times a thread went to sleep\n')
+        self.wfile.write(b'# TYPE acton_rts_sleeps counter\n')
+        for k,v in sorted(data['wt'].items(), key=lambda item: int(item[0])):
+            self.wfile.write(f'acton_rts_sleeps{{thread="{k}"}} {v["sleeps"]}\n'.encode('utf-8'))
+
+        self.wfile.write(b'# HELP acton_rts_conts Execution time of continuations\n')
+        self.wfile.write(b'# TYPE acton_rts_conts histogram\n')
+        for k,v in sorted(data['wt'].items(), key=lambda item: int(item[0])):
+            total['sleeps'] += v['sleeps']
+            for bk in conts_buckets:
```

Comment:
Too much metaprogramming. Better with the upfront style used in `richmon`. It's a little verbose but oh so much easier to read.